File: engine/scripts/lua/tests/test_job.lua
local assert = require("assert")

local function test_job_completion_event()
	init_job_event_logger()
	local eid = spawn_entity()
	local agent_id = spawn_entity()
	-- Add Agent component with skill for "test_job"
	set_component(agent_id, "Agent", { entity_id = agent_id, skills = { test_job = 1.0 } })

	assign_job(
		eid,
		"test_job",
		{ should_fail = false, category = "testing", state = "pending", assigned_to = agent_id }
	)
	local found = false
	for i = 1, 12 do
		run_native_system("JobSystem")
		update_event_buses()
		local events = poll_ecs_event("job_completed")
		if #events > 0 then
			for _, event in ipairs(events) do
				if event.entity == eid then
					found = true
				end
			end
			if found then
				break
			end
		end
	end
	assert.is_true(found, "No job_completed events for this entity")
end

return {
	test_job_completion_event = test_job_completion_event,
}

File: engine_py/tests/test_job.py
def test_job_completion(make_world):
    world = make_world()
    eid = world.spawn_entity()
    world.assign_job(eid, "test_job", category="testing", state="pending")
    found = False
    for _ in range(12):  # enough ticks to guarantee completion
        world.run_native_system("JobSystem")
        world.update_event_buses()
        events = world.poll_ecs_event("job_completed")
        print("Events after tick:", events)
        if any(e["entity"] == eid for e in events):
            found = True
            break
    assert found


def test_get_job_types(make_world):
    world = make_world()

    # Register a custom job type if needed for the test
    def dummy_job_logic(job, progress):
        job["state"] = "complete"
        return job

    world.register_job_type("DummyJob", dummy_job_logic)
    types = world.get_job_types()
    assert isinstance(types, list)
    assert "DummyJob" in types
